solidity/contracts/igps/InterchainGasPaymaster.sol

```diff
@@ -79,25 +110,41 @@ contract InterchainGasPaymaster is IInterchainGasPaymaster, OwnableUpgradeable {
      */
     function quoteGasPayment(uint32 _destinationDomain, uint256 _gasAmount)
         public
-        pure
+        view
         override
         returns (uint256)
     {
-        // Silence compiler warning.
-        _destinationDomain;
-        _gasAmount;
-        // Charge a flat 1 wei fee.
-        // This is an intermediate step toward fully on-chain accurate gas payment quoting.
-        return 1;
+        // Get the gas data for the destination domain.
+        (
+            uint128 _tokenExchangeRate,
+            uint128 _gasPrice
+        ) = getExchangeRateAndGasPrice(_destinationDomain);
+
+        // The total cost quoted in destination chain's native token.
+        uint256 _destinationGasCost = _gasAmount * uint256(_gasPrice);
+
+        // Convert to the local native token.
+        return
+            (_destinationGasCost * _tokenExchangeRate) /
+            TOKEN_EXCHANGE_RATE_SCALE;
     }
 
     /**
-     * @notice Transfers the entire native token balance to the owner of the contract.
-     * @dev The owner must be able to receive native tokens.
+     * @notice Gets the token exchange rate and gas price from the configured gas oracle
+     * for a given destination domain.
+     * @param _destinationDomain The destination domain.
+     * @return tokenExchangeRate The exchange rate of the remote native token quoted in the local native token.
+     * @return gasPrice The gas price on the remote chain.
      */
-    function claim() external {
-        // Transfer the entire balance to owner.
-        (bool success, ) = owner().call{value: address(this).balance}("");
-        require(success, "!transfer");
+    function getExchangeRateAndGasPrice(uint32 _destinationDomain)
```

Comment:
do we need to worry about staleness here?

Reply:
Not 100% sure what you mean by stale but the assumption is that the gas oracle that's called into will be accurate (or if it's not accurate, because the IGP set that gas oracle, the IGP is aware that it may not be super accurate)